internal/handlers/common/add_fields_iterator.go

```diff
@@ -65,6 +87,58 @@ func (iter *addFieldsIterator) Close() {
 	iter.iter.Close()
 }
 
+// processAddFieldsError takes internal error related to operator evaluation and
+// returns proper CommandError that can be returned by $addFields aggregation stage.
+func processAddFieldsError(err error) error {
+	if err == nil {
+		return nil
+	}
+
+	var opErr operators.OperatorError
+
+	if !errors.As(err, &opErr) {
+		return err
+	}
+
+	switch opErr.Code() {
```

Comment:
Please add tests for checking error message for each of new errors please :pray:
We should be able to check the message for `ErrTooManyFields`, `ErrArgsInvalidLen` `ErrInvalidExpression`  and `ErrInvalidNestedExpression`.

Reply:
The only error not tested from this list yet is ErrInvalidExpression, adding test for it.

Reply:
Yep thanks!